packages/@lwc/engine-core/src/framework/mutation-tracker.ts

```diff
@@ -28,11 +28,37 @@ export function componentValueMutated(vm: VM, key: PropertyKey) {
     }
 }
 
-export function componentValueObserved(vm: VM, key: PropertyKey) {
+export function componentValueObserved(vm: VM, key: PropertyKey, target: any = {}) {
     // On the server side, we don't need mutation tracking. Skipping it improves performance.
     if (process.env.IS_BROWSER) {
         valueObserved(vm.component, key);
     }
+
+    // Putting this here for now, the idea is to subscribe to a signal when there is an active template reactive observer.
+    // This would indicate that:
+    //      1. The template is currently being rendered
+    //      2. There was a call to a getter bound to the LWC class
+    // With this information we can infer that it is safe to subscribe the re-render callback to the signal, which will
+    // mark the VM as dirty and schedule rehydration.
+    if (
+        target &&
+        typeof target === 'object' &&
+        'value' in target &&
+        'subscribe' in target &&
+        typeof target.subscribe === 'function'
+    ) {
```

Comment:
These conditions look right to me for now. Sometime after this PR lands, we may want to replace the last three conditions with something like `&& target.isSignal === SUPER_SECRET_SYMBOL`, where `const SUPER_SECRET_SYMBOL === Symbol("SymbolSigil")`. This would allow us to lock down initial usage of the signal machinery by restricting who has access to that symbol in the runtime (i.e. so that customers can't start using signals before we're ready).